ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -852,3 +852,174 @@ func TestVirtual_DeduplicateCallAfterDeactivation_PrevVE(t *testing.T) {
+				require.Equal(t, int32(0), report.UnorderedPendingCount)
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Nil(t, report.ProvidedContent)
+				gotResult <- struct{}{}
+				return false
+			})
+			typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
+				require.Equal(t, objectRef, response.Callee)
+				require.NotEmpty(t, response.ResponseDelegationSpec)
+				require.Equal(t, objectRef, response.ResponseDelegationSpec.Callee)
+				require.Equal(t, server.GlobalCaller(), response.ResponseDelegationSpec.DelegateTo)
+
+				// send delegation request finished with deactivate flag
+				pl := payload.VDelegatedRequestFinished{
+					Callee:       objectRef,
+					CallOutgoing: outgoing,
+					CallFlags:    payload.BuildCallFlags(deactivateIsolation.Interference, deactivateIsolation.State),
+					LatestState: &payload.ObjectState{
+						State:       nil,
+						Deactivated: true,
+					},
+				}
+				server.SendPayload(ctx, &pl)
+				server.WaitActiveThenIdleConveyor()
+				return false
+			})
+
+			{ // create object with pending deactivation
+				report := payload.VStateReport{
+					Status:                      payload.Ready,
+					AsOf:                        p1,
+					Object:                      objectRef,
+					OrderedPendingCount:         1,
+					OrderedPendingEarliestPulse: p1,
+					LatestDirtyState:            stateRef,
+					ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+						LatestDirtyState: &payload.ObjectState{
+							Reference: stateRef.GetLocal(),
+							Class:     class,
+							State:     []byte(origMem),
+						},
+						LatestValidatedState: &payload.ObjectState{
+							Reference: stateRef.GetLocal(),
+							Class:     class,
+							State:     []byte(origMem),
+						},
+					},
+				}
+				server.SendPayload(ctx, &report)
+				server.WaitActiveThenIdleConveyor()
+			}
+			{ // fill object pending table
+				dcr := payload.VDelegatedCallRequest{
+					Callee:       objectRef,
+					CallOutgoing: outgoing,
+					CallFlags:    payload.BuildCallFlags(deactivateIsolation.Interference, deactivateIsolation.State),
+				}
+				server.SendPayload(ctx, &dcr)
+				server.WaitActiveThenIdleConveyor()
+			}
+
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+
+			{
+				pl := payload.VCallRequest{
+					CallType:            payload.CTMethod,
+					CallFlags:           payload.BuildCallFlags(testCase.isolation.Interference, testCase.isolation.State),
+					Caller:              server.GlobalCaller(),
+					Callee:              objectRef,
+					CallSiteDeclaration: class,
+					CallSiteMethod:      "Check",
+					CallOutgoing:        checkOutgoing,
+				}
+				server.SendPayload(ctx, &pl)
+			}
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, oneExecutionEnded)
+			server.IncrementPulseAndWaitIdle(ctx)
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, gotResult)
```

Comment:
Not custom channel, use `commontestutils.WaitSignalsTimed(t, 10*time.Second, typedChecker.VStateReport.Wait(ctx, 1))` instead

Reply:
thanks a lot, done